### ISSUE TITLE
Dependencies

### ISSUE THREAD

**Issue by saguillo2000** [MathieuCarriere/difftda#2]
Hi Mathieu,

I'm trying to test the code you created from the article you got inspired. The problem is that if I use the last version of Ghudi, 3.4.1, I can't install Tensorflow 2.4.1 as  Guhdi is supported in python 3.9.

I tried the two following approaches without exit:

1st ) Installed last version of Guhdi 3.4.1 and Tensorflow 2.7.0, it gives me the following error, (I suspect this is for the Tensorflow version):

![image](https://user-images.githubusercontent.com/47833532/146850348-3898d4eb-b80d-401e-b990-858f33baa24d.png)

2ns) Installed Gudhi 3.1.1 with Tensorflow 2.4.1, it gives me the following error in imports:

![image](https://user-images.githubusercontent.com/47833532/146850447-0f8bf1ed-db90-4452-aeaf-991eff096780.png)

I list the last conda list to know if is something with a wrong version:

# Name                    Version                   Build  Channel
absl-py                   0.15.0                   pypi_0    pypi
argon2-cffi               21.3.0                   pypi_0    pypi
argon2-cffi-bindings      21.2.0                   pypi_0    pypi
astunparse                1.6.3                    pypi_0    pypi
attrs                     21.2.0                   pypi_0    pypi
backcall                  0.2.0                    pypi_0    pypi
bleach                    4.1.0                    pypi_0    pypi
boost-cpp                 1.72.0               h0caebb8_0    conda-forge/label/cf202003
bzip2                     1.0.8                hfa6e2cd_2    conda-forge/label/cf202003
ca-certificates           2019.11.28           hecc5488_0    conda-forge/label/cf202003
certifi                   2019.11.28       py38h32f6830_1    conda-forge/label/cf202003
cffi                      1.15.0                   pypi_0    pypi
cgal-cpp                  5.0.1                h26c6e29_1    conda-forge/label/cf202003
charset-normalizer        2.0.9                    pypi_0    pypi
click                     8.0.3                    pypi_0    pypi
colorama                  0.4.4                    pypi_0    pypi
cycler                    0.11.0                   pypi_0    pypi
cython                    0.29.26                  pypi_0    pypi
debugpy                   1.5.1                    pypi_0    pypi
decorator                 5.1.0                    pypi_0    pypi
defusedxml                0.7.1                    pypi_0    pypi
eigen                     3.3.7             he980bc4_1001    conda-forge/label/cf202003
entrypoints               0.3                      pypi_0    pypi
flatbuffers               1.12                     pypi_0    pypi
fonttools                 4.28.5                   pypi_0    pypi
gast                      0.3.3                    pypi_0    pypi
google-auth-oauthlib      0.4.6                    pypi_0    pypi
google-pasta              0.2.0                    pypi_0    pypi
grpcio                    1.32.0                   pypi_0    pypi
gudhi                     3.1.1            py38h347c182_1    conda-forge/label/cf202003
h5py                      2.10.0                   pypi_0    pypi
importlib-metadata        4.10.0                   pypi_0    pypi
importlib-resources       5.4.0                    pypi_0    pypi
ipykernel                 6.6.0                    pypi_0    pypi
ipython                   7.30.1                   pypi_0    pypi
ipython-genutils          0.2.0                    pypi_0    pypi
ipywidgets                7.6.5                    pypi_0    pypi
itsdangerous              2.0.1                    pypi_0    pypi
jedi                      0.18.1                   pypi_0    pypi
jinja2                    3.0.3                    pypi_0    pypi
joblib                    1.1.0                    pypi_0    pypi
jsonschema                4.3.2                    pypi_0    pypi
jupyter                   1.0.0                    pypi_0    pypi
jupyter-client            7.1.0                    pypi_0    pypi
jupyter-console           6.4.0                    pypi_0    pypi
jupyter-core              4.9.1                    pypi_0    pypi
jupyterlab-pygments       0.1.2                    pypi_0    pypi
jupyterlab-widgets        1.0.2                    pypi_0    pypi
keras-preprocessing       1.1.2                    pypi_0    pypi
kiwisolver                1.3.2                    pypi_0    pypi
libblas                   3.8.0           7_h8933c1f_netlib    conda-forge/label/cf202003
libcblas                  3.8.0           7_h8933c1f_netlib    conda-forge/label/cf202003
liblapack                 3.8.0           7_h8933c1f_netlib    conda-forge/label/cf202003
m2w64-gcc-libgfortran     5.3.0                         6    conda-forge
m2w64-gcc-libs            5.3.0                         7    conda-forge
m2w64-gcc-libs-core       5.3.0                         7    conda-forge
m2w64-gmp                 6.1.0                         2    conda-forge
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    conda-forge
markdown                  3.3.6                    pypi_0    pypi
markupsafe                2.0.1                    pypi_0    pypi
matplotlib                3.5.1                    pypi_0    pypi
matplotlib-inline         0.1.3                    pypi_0    pypi
mistune                   0.8.4                    pypi_0    pypi
mpfr                      4.0.2                h2fa13f4_0    conda-forge/label/cf202003
mpir                      3.0.0             he025d50_1002    conda-forge/label/cf202003
msys2-conda-epoch         20160418                      1    conda-forge
nbclient                  0.5.9                    pypi_0    pypi
nbconvert                 6.3.0                    pypi_0    pypi
nbformat                  5.1.3                    pypi_0    pypi
nest-asyncio              1.5.4                    pypi_0    pypi
notebook                  6.4.6                    pypi_0    pypi
numpy                     1.19.5                   pypi_0    pypi
oauthlib                  3.1.1                    pypi_0    pypi
openssl                   1.1.1e               hfa6e2cd_0    conda-forge/label/cf202003
opt-einsum                3.3.0                    pypi_0    pypi
packaging                 21.3                     pypi_0    pypi
pandas                    1.3.5                    pypi_0    pypi
pandocfilters             1.5.0                    pypi_0    pypi
parso                     0.8.3                    pypi_0    pypi
pickleshare               0.7.5                    pypi_0    pypi
pillow                    8.4.0                    pypi_0    pypi
pip                       20.0.2                     py_2    conda-forge/label/cf202003
prometheus-client         0.12.0                   pypi_0    pypi
prompt-toolkit            3.0.24                   pypi_0    pypi
pyasn1                    0.4.8                    pypi_0    pypi
pycparser                 2.21                     pypi_0    pypi
pygments                  2.10.0                   pypi_0    pypi
pyparsing                 3.0.6                    pypi_0    pypi
pyrsistent                0.18.0                   pypi_0    pypi
python                    3.8.2           h5fd99cc_4_cpython    conda-forge/label/cf202003
python-dateutil           2.8.2                    pypi_0    pypi
python_abi                3.8                      1_cp38    conda-forge/label/cf202003
pytz                      2021.3                   pypi_0    pypi
pywin32                   303                      pypi_0    pypi
pywinpty                  1.1.6                    pypi_0    pypi
pyzmq                     22.3.0                   pypi_0    pypi
qtconsole                 5.2.2                    pypi_0    pypi
qtpy                      1.11.3                   pypi_0    pypi
requests                  2.26.0                   pypi_0    pypi
requests-oauthlib         1.3.0                    pypi_0    pypi
scikit-learn              1.0.1                    pypi_0    pypi
scipy                     1.7.3                    pypi_0    pypi
send2trash                1.8.0                    pypi_0    pypi
setuptools                46.0.0           py38h32f6830_2    conda-forge/label/cf202003
six                       1.15.0                   pypi_0    pypi
sklearn                   0.0                      pypi_0    pypi
sqlite                    3.30.1               hfa6e2cd_0    conda-forge/label/cf202003
tensorboard               2.7.0                    pypi_0    pypi
tensorboard-data-server   0.6.1                    pypi_0    pypi
tensorboard-plugin-wit    1.8.0                    pypi_0    pypi
tensorflow                2.4.1                    pypi_0    pypi
tensorflow-addons         0.15.0                   pypi_0    pypi
tensorflow-estimator      2.4.0                    pypi_0    pypi
termcolor                 1.1.0                    pypi_0    pypi
terminado                 0.12.1                   pypi_0    pypi
testpath                  0.5.0                    pypi_0    pypi
threadpoolctl             3.0.0                    pypi_0    pypi
tornado                   6.1                      pypi_0    pypi
traitlets                 5.1.1                    pypi_0    pypi
typeguard                 2.13.3                   pypi_0    pypi
typing-extensions         3.7.4.3                  pypi_0    pypi
ucrt                      10.0.20348.0         h57928b3_0    conda-forge
vc                        14.2                 hb210afc_5    conda-forge
vs2015_runtime            14.29.30037          h902a5da_5    conda-forge
wcwidth                   0.2.5                    pypi_0    pypi
webencodings              0.5.1                    pypi_0    pypi
wheel                     0.37.0                   pypi_0    pypi
widgetsnbextension        3.5.2                    pypi_0    pypi
wincertstore              0.2                   py38_1003    conda-forge/label/cf202003
wrapt                     1.12.1                   pypi_0    pypi
zipp                      3.6.0                    pypi_0    pypi
zlib                      1.2.11                   vc14_0  [vc14]  conda-forge/label/cf202003

**Comment by KKMGeraedts** [comment]
Hi, 

I am also trying to test the code and I am running into the same problem as mentioned above. Since the issue has been closed I was wondering if a solution has been found?

Thanks,
Karel

**Comment by saguillo2000** [comment]
Hi @KKMGeraedts!

Yes, I could run the code. I changed small fixes in the difftda.py because of the TensorFlow version. I'm updating the project in a fork I have in my account: https://github.com/saguillo2000/difftda

I use Windows 10, so maybe there is a slight change in some versions, if you have any problem I'll try to help.

Oriol
